papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -408,86 +422,100 @@ For example, a **signac** data space that describes weather data for multiple ci
-However, replacing individual JSON files as the primary data source for **signac** without breaking **signac**'s API would require a generic means for providing the same interface to the underlying index and metadata irrespective of the underlying storage mechanism.
-Once developed, however, such an API would abstract out enough of the internals of **signac** to enable other generalizations as well, such as making it relatively easy to support alternate (and nearly arbitrary) data space layouts.
+While attempting to overcome this hurdle, we identified a core problem that our current API for modifying JSON files was solving in a limited manner: namely, the creation of a dictionary-like interface to an underlying resource.
+Numerous other well-known Python packages (h5py and Zarr immediately come to mind) also use dictionary-like interfaces to make working with complex resources feel natural to Python users.
+Most such packages implement this layer directly for their particular use case, but the nature of the problem suggested to us the possibility of developing a more generic representation of this interface.
+Indeed, the Python standard library's ``collections.abc`` module is designed exactly for the purpose of making it easy to define objects that "look like" standard Python objects while having completely customizable behavior under the hood.
+As such, we saw an opportunity to specialize this pattern for a specific use case: the transparent synchronization of a Python object with an underlying resource.
 
-The synced collections subpackage of **signac** represents the culmination of our efforts to expose this functionality, providing a generic framework within which interfaces corresponding to any of Python's built-in types can be easily constructed with arbitrary underlying synchronization protocols.
-For instance, with synced collections it becomes easy to define a new list-like type that automatically saves all its data in a plain-text CSV format.
-However, the flexibility of this new framework extends far beyond that, defining a generic protocol that can be used to provide a dictionary, list, or set-like API to any arbitrary underlying data structure, including other in-memory objects that do not present a similarly Pythonic API.
+The **synced collections** framework represents the culmination of our efforts in this direction, providing a generic framework within the interfaces of any abstract data type can be mapped to arbitrary underlying synchronization protocols.
+In **signac**, this framework allows us to hide the details of a particular file storage medium (like JSON) behind a dictionary-like interface, but it can just as easily be used for tasks such as creating a set-like interface to an underlying extension type or wrapping a directory manager in a list-like interface.
```

Comment:
```suggestion
In **signac**, this framework allows us to hide the details of a particular file storage medium (like JSON) behind a dictionary-like interface, but it can just as easily be used for tasks such as creating a new, list-like interface that automatically saves all its data in a plain-text CSV format.
```

> but it can just as easily be used for tasks such as creating a set-like interface to an underlying extension type or wrapping a directory manager in a list-like interface

These examples seemed a little abstract. The previous example with CSV files was easier to understand because it used an extension and data type that were common and familiar. Was there a specific reason for replacing the previous example with more general ones?

Reply:
I think to mention that sets could effectively be added easily.

Reply:
Ah, is that something that is normally hard to do?

Reply:
I would be fine with dropping the `set` example but I do like the concreteness of CSV. Feels more tangible than a directory manager.